lib/chef/provisioning/aws_driver/aws_resource.rb

```diff
@@ -67,6 +69,13 @@ def aws_object
     raise NotImplementedError, :aws_object
   end
 
+  def aws_object_id
+    @aws_object_id ||= begin
+      o = aws_object
+      o.public_send(self.class.aws_sdk_class_id) if o
```

Comment:
We do this `public_send` thing so often it seems like the classes should have an instance method that does it. (Not that I'm recommending it here.)